ddtrace/appsec/ddwaf/__init__.py

```diff
@@ -126,33 +139,34 @@ def run(
             # type: (...) -> DDWaf_result
             start = time.time()
 
-            ctx = ddwaf_context_init(self._handle)
-            if ctx == 0:
-                LOGGER.error("DDWaf failure to create the context")
+            if self._ctx == 0:
+                LOGGER.warning("DDWaf failsafe to create the context")
+                self._ctx = ddwaf_context_init(self._handle)
+
+            if self._ctx == 0:
+                LOGGER.error("DDWaf failure: no context created")
                 return DDWaf_result(None, [], 0, (time.time() - start) * 1e6)
 
+            result = ddwaf_result()
```

Comment:
This block was previously inside a try block, are we sure about removing that?

Reply:
Yes, the try/finally was only present to free the context. We don't want that anymore.